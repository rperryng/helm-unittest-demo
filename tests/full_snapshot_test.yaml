suite: Full Snapshot
templates:
  - '*.yaml'
  - dash-case-filename.yaml
tests:
  - it: matches the snapshot
    asserts:
      - matchSnapshot: {}
