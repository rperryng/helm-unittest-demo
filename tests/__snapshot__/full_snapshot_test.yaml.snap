matches the snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: helm-demo
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: helm-demo-0.1.0
      name: RELEASE-NAME-helm-demo
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: helm-demo
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: helm-demo
        spec:
          containers:
            - image: nginx:1.16.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: helm-demo
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-helm-demo
  2: |
    apiVersion: autoscaling/v2beta1
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: helm-demo
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: helm-demo-0.1.0
      name: RELEASE-NAME-helm-demo
    spec:
      maxReplicas: 100
      metrics:
        - resource:
            name: cpu
            targetAverageUtilization: 80
          type: Resource
      minReplicas: 1
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME-helm-demo
  3: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: helm-demo
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: helm-demo-0.1.0
      name: RELEASE-NAME-helm-demo
    spec:
      rules:
        - host: chart-example.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-helm-demo
                    port:
                      number: 80
                path: /
                pathType: ImplementationSpecific
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: helm-demo
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: helm-demo-0.1.0
      name: RELEASE-NAME-helm-demo
    spec:
      ports:
        - name: http
          port: 80
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: helm-demo
      type: ClusterIP
  5: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: helm-demo
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: helm-demo-0.1.0
      name: RELEASE-NAME-helm-demo
